community/modules/scripts/wait-for-startup/scripts/wait-for-startup-status.sh

```diff
@@ -28,11 +28,15 @@ fi
 
 now=$(date +%s)
 deadline=$(("${now}" + "${TIMEOUT}"))
+error_file=$(mktemp)
 
 until [ "${now}" -gt "${deadline}" ]; do
-	GCLOUD="gcloud compute instances get-serial-port-output ${INSTANCE_NAME} --port 1 --zone ${ZONE} --project ${PROJECT_ID}"
+	ser_log=$(gcloud compute instances get-serial-port-output "${INSTANCE_NAME}" --port 1 --zone "${ZONE}" --project "${PROJECT_ID}" 2>"${error_file}") || {
```

Comment:
We are now storing the full output, instead of immediately feeding it to grep. IDK anything about shell variable buffers. The serial port output can be very long. Should we be at all concerned about storing this in a variable?

If this concern is valid then ideally we would want to do the grep at this point. I think you could do it by performing the grep up front and then just checking the error file instead of using the OR. If you know this to not be a concern then respond and close.